packages/@cdktf/hcl2cdk/lib/generation.ts

```diff
@@ -59,6 +62,75 @@ function getReference(graph: DirectedGraph, id: string) {
   }
 }
 
+function destructureAst(ast: t.Expression): string[] | undefined {
+  switch (ast.type) {
+    case "Identifier":
+      return [ast.name];
+    case "MemberExpression":
+      const object = destructureAst(ast.object);
+      const property = destructureAst(ast.property as t.Expression);
+      if (object && property) {
+        return [...object, ...property];
+      } else {
+        return undefined;
+      }
+    default:
+      return undefined;
+  }
+}
+
+function findTypeOfReference(
+  scope: Scope,
+  ast: t.Expression,
+  references: Reference[]
+): AttributeType {
+  const isReferenceWithoutTemplateString =
+    ast.type === "MemberExpression" &&
+    ast.object.type === "Identifier" &&
+    references.length === 1;
+  // If we only have one reference this is a
+  if (isReferenceWithoutTemplateString) {
+    if (references[0].isVariable) {
+      return "dynamic";
+    }
+
+    const destructuredAst = destructureAst(ast);
+    if (!destructuredAst) {
+      logger.debug(
+        `Could not destructure ast: ${JSON.stringify(ast, null, 2)}`
+      );
+      return "dynamic";
+    }
+
+    const [astVariableName, ...attributes] = destructuredAst;
+    const variable = Object.values(scope.variables).find(
+      (x) => x.variableName === astVariableName
+    );
+
+    if (!variable) {
+      // We don't know, this should not happen, but if it does we assume the worst case and make it dynamic
+      return "dynamic";
```

Comment:
I'd prefer failing if this should not happen – or at least logging an error.

Reply:
We don't usually fail with ambigous content, so I would not like to add this here. I can add a debug log though